Lesson18/nurbasak/recipeBook.js

```diff
@@ -0,0 +1,243 @@
+- If not found, shows a message.
+*/
+
+console.log(" ");
+console.log("STEP 4 : VIEW A RECIPE BY NAME.");
+console.log(" ");
+
+function viewRecipe(name) {
+  for(let i=0; i<recipes.length;i++) {
+    if(recipes[i].name===name) {
+      console.log(recipes[i]);
+      return;
+    } 
+  } 
+  console.log(`Recipe with name ${name} not found.`);
+}
+
+viewRecipe('Minestrone Soup');
+viewRecipe('Pasta');
+
+/* When I log STEP 4, it looks in terminal with curly brackets and square brackets as well. I couldn't find out why. */
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Recipe
+-----------------------------------------------------------
+Function: updateRecipe(name, newIngredients, newCookingTime)
+- Finds the recipe by name.
+- Updates ingredients and cookingTime.
+- Logs success or error message.
+*/
+console.log(" ");
+console.log("STEP 5 : UPDATE A RECIPE.");
+console.log(" ");
+function updateRecipe(name,newIngredients,newCookingTime) {
+for(let i = 0; i < recipes.length; i++){
+  if(recipes[i].name === name){
+    recipes[i].ingredients = newIngredients;
+    recipes[i].cookingTime = newCookingTime;
+    console.log( `${name} recipe is updated.`);
+    return;
+  } 
+} 
+console.log(`Recipe with name ${name} is not found.`);
+}
+
+console.log(" ");
+
+updateRecipe('Arancini', ['Unsalted Butter', 'Garlic', 'Rissotto Rice', 'Stock', 'Parmesan', 'Flour', 'Eggs', 'Breadcrumbs', 'Olive Oil', 'Mozarella'], 45);
+updateRecipe('Pasta', ['Tomatoes ', 'Red Onion ', 'Garlic ', 'Basil ', ' Olive Oil ', 'Balsamic Vinegar ', 'Baguette '], 20);
+
+/*
+-----------------------------------------------------------
+  STEP 6: Delete a Recipe
+-----------------------------------------------------------
+Function: deleteRecipe(name)
+- Finds and removes the recipe from the array.
+- Logs success or error message.
+*/
+
+console.log(" ");
+console.log("STEP 6 : DELETE A RECIPE.");
+console.log(" ");
+
+function deleteRecipe(name){
+  for ( let i = 0; i<recipes.length; i++){
+    if(recipes[i].name === name){
+      recipes.splice(i, 1);
+      console.log(`${name} recipe deleted.`);
+      return;
+    }
+  } console.log(`${name} recipe is not found.` );
+}
+
+deleteRecipe('Caprese Salad');
+deleteRecipe('Pasta');
+
+/*
+-----------------------------------------------------------
+  STEP 7: Extra Challenge – Filter Recipes
+-----------------------------------------------------------
+Function: filterByIngredient(ingredient)
+- Shows all recipes that use a certain ingredient.
+
+Function: filterByMaxTime(maxMinutes)
+- Shows recipes that take <= maxMinutes to cook.
+*/
+console.log(" ");
+console.log("STEP 7 : FILTER RECIPES BY INGREDIENT.");
+console.log(" ");
+
+function filterByIngredient(ingredient) {
+  console.log(`Recipes with ingredient: ${ingredient}`);
+  for (let i = 0; i < recipes.length; i++) {
+    if (recipes[i].ingredients.includes(ingredient)) {
+      console.log (recipes[i].name);
+      return;
```

Comment:
The function only finds the first recipe with matching ingredient, then stops. Remove 'return' keyword and instead add a state (e.g. counter var and increment the count with every recipe found).